graphql_compiler/schema/__init__.py

```diff
@@ -296,7 +296,7 @@ def _serialize_date(value):
     return value.isoformat()
 
 
-def _parse_date_value(value):
+def _parse_date_value(value: str) -> date:
```

Comment:
nit: I would use `Any` instead of `str` for two reasons:
1. To be consistent with the `_parse_datetime_value` function and all of the GraphQL core next parsing functions. (These GraphQL core type annotation are also probably in turn derived from the GraphQL js library). 
2.  The input here is likely going to be a JSON value and the type of this value might be an union of the possible JSON types. In that case, mypy might raise an error unless we use a `cast`.

Reply:
The third and most important reason is that I would like to avoid future discussions over the parsing function type annotations and seems like the best way to do that is to be consistent from the start.

Reply:
Yeah, you're right — the proper type should be `Any`.
https://github.com/graphql-python/graphql-core/blob/master/src/graphql/type/scalars.py#L100
https://github.com/graphql-python/graphql-core/blob/master/src/graphql/type/scalars.py#L64